README.md

```diff
@@ -1,116 +1,219 @@
 cp /path/to/update_default_resources.c .
-# build all the Ubuntu 20 packages for amd64 using the docker container created above
-./build-env/bin/docker-run.sh pelion-focal-source ./build-env/bin/pelion-build-all.sh --deps --install --build --source --arch=amd64
+```
+
+2. Run build scripts:
+```bash
+# build all the Ubuntu 20 packages for amd64 using docker container
+./build-env/bin/docker-run-env.sh focal ./build-env/bin/build-all.sh --deps --install --build --source --arch=amd64
 # deb packages will be available in ./build/deploy/deb/focal/main/
 
 
 # then if you'd like to rebuild an individual package (ex: mbed-edge-core-devmode)
-./build-env/bin/docker-run.sh pelion-focal-source ./mbed-edge-core-devmode/deb/build.sh --install --build --source --arch=amd64
+./build-env/bin/docker-run-env.sh focal ./mbed-edge-core-devmode/deb/build.sh --install --build --source --arch=amd64
 ```
 
-## Build environment
+Alternative way to run build scripts:
 
-1. Scripts for creating clean build docker images:
-```
-$ ./build-env/bin/docker-ubuntu-bionic-create.sh # Ubuntu 18.04
+```bash
+# build all the Ubuntu 20 packages for amd64 using docker container
+./build-env/bin/build-all.sh --docker=focal --deps --install --build --source --arch=amd64
+# deb packages will be available in ./build/deploy/deb/focal/main/
+
+
+# then if you'd like to rebuild an individual package (ex: mbed-edge-core-devmode)
+./mbed-edge-core-devmode/deb/build.sh --docker=focal --install --build --source --arch=amd64
 ```
 
-The  above  script  creates   docker  images  `pelion-bionic-build`  with  build
-essentials  and `pelion-bionic-source`  with  additional  packages required  for
-generating source packages (for example git, npm, python).
 
-To build packages for other distributions, replace `pelion-bionic` with the name
-of the distribution.  For example, to build for Debian 10, use `debian-buster`.
+## Selecting target distribution with `--docker` switch
 
-To add prefix to docker images export `PELION_DOCKER_PREFIX`  variable  in  your
-shell with your prefix:
+The  `--docker` switch  (or short  `-d`) is  used to  select build  environment.
+Environment  configurations   are  stored  in  `./build-env/target/`.   To  list
+available environments run:
+```bash
+./build-env/bin/build-all.sh -l env
 ```
-export PELION_DOCKER_PREFIX=${USER}-
+
+The `-d`  accepts also  partial environments  names (`-d  rh`, `-d  rhel/8`, `-d
+rhel-8`,  `-d  rhel`  -  all  are  valid).  When  name  matches  more  than  one
+environment, script will print error:
+```bash
+$ ./build-env/bin/build-all.sh -d 8
+Unable to load environment: ambiguous environment name, matches:
+centos/8 rhel/8
+
 ```
 
-The system is configured to use `sudo` without a password.
+## Preparing host to compile packages
+### System requirements
+The `build-all.sh` Script requires docker, bash 4.2+ and gnu-getopt.
 
+To  build  arm64  packages for  *Red  Hat*  or  *Centos*  `qemu` is  required  and
+qemu-docker integration.
```

Comment:
Can we add: "because RedHat/CentOS don't support cross compiling" ?

Reply:
Done